start.sh

```
@@ -50,16 +50,37 @@ if [ $ATTEMPTS -ge $MAX_ATTEMPTS ]; then
-# Wait for Vespa to be ready
+# Wait for Vespa to be ready. Vespa is optional by default so a model download
+# or search deployment issue does not prevent the app from starting.
+VESPA_REQUIRED="${VESPA_REQUIRED:-false}"
+VESPA_ATTEMPTS=0
+VESPA_MAX_ATTEMPTS="${VESPA_WAIT_MAX_ATTEMPTS:-30}"
+VESPA_READY=false
+
 echo "Waiting for Vespa config server..."
-until curl -f http://${VESPA_HOST:-vespa}:19071/state/v1/health 2>/dev/null; do
-  echo "Vespa config server is unavailable - sleeping"
+while [ $VESPA_ATTEMPTS -lt $VESPA_MAX_ATTEMPTS ]; do
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)